### ISSUE TITLE
`VisualStudioIntegration.IsVisualStudioRunning` can dead lock the app

### ISSUE THREAD

**Issue by RussKie** [gitextensions/gitextensions#10856]
### Environment

- Git Extensions 4.0.2.16100
- Build 25100ec1f7c8da8f798613da74826038af81a50e
- Git 2.39.2.windows.1
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.15
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 8.0.0-preview.2.23128.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Issue description

Invoking the context menu that contains "Open in Visual Studio" menu can dead lock the app and the only option is to hard kill the process.
https://github.com/gitextensions/gitextensions/blob/037cb04d3b1168d197643512b5bd2d1b506042e1/GitUI/UserControls/FileStatusList.cs#L1412

I've taken a memdump:
![image](https://user-images.githubusercontent.com/4403806/229668411-452b3f9f-5037-4730-bdb6-c3f44d87cd4f.png)

### Steps to reproduce

It's quite intermittent but I observed this issue few times over the months. In essence, it occurs whenever I right click on files in FormCommit dialog that I want to stage or reset.

**Comment by RussKie** [comment]
I suspect that this is happening is there's an open VS instance that's locked itself (e.g., being a debug state or frozen for some other reason). This blocks all/some of the COM calls used in `VisualStudioIntegration`, which leads to the app's dead lock.

**Comment by gerhardol** [comment]
Fixed in #10887
